addons/website/tests/test_configurator_translation.py

```diff
@@ -0,0 +1,64 @@
+class TestConfigurator(odoo.tests.HttpCase):
+
+    def _theme_upgrade_upstream(self):
+        # patch to prevent module install/upgrade during tests
+        pass
+
+    def setUp(self):
+        super().setUp()
+
+        def iap_jsonrpc_mocked_configurator(*args, **kwargs):
+            endpoint = args[0]
+            params = kwargs.get('params', {})
+            language = params.get('lang', 'en_US')
+            if endpoint.endswith('/api/website/1/configurator/industries'):
+                if language == 'fr_FR':
+                    return {"industries": [
+                        {"id": 1, "label": "abbey in fr"},
+                        {"id": 2, "label": "aboriginal and torres strait islander organisation in fr"},
+                        {"id": 3, "label": "aboriginal art gallery in fr"},
+                        {"id": 4, "label": "abortion clinic in fr"},
+                        {"id": 5, "label": "abrasives supplier in fr"},
+                        {"id": 6, "label": "abundant life church in fr"}]}
+                else:
+                    return {"industries": [
+                        {"id": 1, "label": "abbey"},
+                        {"id": 2, "label": "aboriginal and torres strait islander organisation"},
+                        {"id": 3, "label": "aboriginal art gallery"},
+                        {"id": 4, "label": "abortion clinic"},
+                        {"id": 5, "label": "abrasives supplier"},
+                        {"id": 6, "label": "abundant life church"}]}
+            elif '/api/website/2/configurator/recommended_themes' in endpoint:
+                return []
+            elif '/api/website/2/configurator/custom_resources/' in endpoint:
+                return {'images': {}}
+
+            iap_jsonrpc_mocked()
+
+        with mute_logger('odoo.addons.base.models.ir_translation'):
+            self.env["base.language.install"].create({'lang': 'fr_FR', 'overwrite': True}).lang_install()
+        self.lang_id = self.env.ref('base.lang_fr')
+        self.fr_website_id = self.env['website'].create({'name': "Website Fr", 'default_lang_id': self.lang_id.id})
+        iap_patch = patch('odoo.addons.iap.tools.iap_tools.iap_jsonrpc', iap_jsonrpc_mocked_configurator)
+        iap_patch.start()
+        self.addCleanup(iap_patch.stop)
+
+        patcher = patch('odoo.addons.website.models.ir_module_module.IrModuleModule._theme_upgrade_upstream', wraps=self._theme_upgrade_upstream)
+        patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_01_configurator_translation(self):
+        # disable configurator todo to ensure this test goes through
+        active_todo = self.env['ir.actions.todo'].search([('state', '=', 'open')], limit=1)
+        active_todo.update({'state': 'done'})
+        self.start_tour('/website/force/%s?path=%%2Fwebsite%%2Fconfigurator%%3Fdebug%%3Dassets' % self.fr_website_id.id, 'configurator_translation', login='admin')
```

Comment:
The test/tour is not crashing without your first commit (about infinite loading).
Do you think you can call a second tour in the same test method to check this case? (which was the only really blocking issue)

Reply:
Ok so I was thinking : the second commit prevent the bug from ever happening. We're always going to have a frontend_lang equal to the website we're going to edit in the second commit because, when we create a website, we go on the configurator page before we can skip it. And going to the configurator sets the correct frontend_lang.

That doesn't mean the first commit is useless however, in fact it should be pack-ported because the issue also happens in 14.0:
 
 - Create a new databse
 - Install website, and add french on website 1
 - Change lang of website 1 to french while visiting
 - Create a new website with english as a language
 - Select a theme
 - The editor won't start even though ?enable_editor=1 is in the address bar

Should I make a PR for 14.0? I'll add tour that reproduce the steps in both PR..

Reply:
If you can then extract that commit in 14.0 that would be perfect.
It will be added in 15.0 and master during forward port, not needed in this PR then IMO